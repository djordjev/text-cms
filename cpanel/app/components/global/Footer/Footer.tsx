export interface FooterProps {}

const Footer = () => {
  return (
    <footer className="u-footer u-footer-center u-p-2x u-bg-base-300 u-text-base-content">
      <aside className="u-block">
        <p className="u-inline-block">Copyright Â© 2024 - Djordje Vukovic</p>
        <div className="u-mx-2xs u-inline">/</div>
        <a
          href="https://github.com/djordjev/text-cms"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </aside>
    </footer>
  );
};

export { Footer };
