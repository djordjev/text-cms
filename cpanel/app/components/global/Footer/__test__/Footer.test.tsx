import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { Footer, FooterProps } from '../Footer';

test('Footer', () => {
  const props: FooterProps = {};

  const TestComponent = createRemixStub([
    {
      path: '/',
      Component: () => <Footer {...props} />
    }
  ]);

  const { container } = render(<TestComponent />);

  expect(screen.getByText(/Djordje Vukovic/)).toBeInTheDocument();
  expect(screen.getByRole('link', { name: 'GitHub' })).toHaveAttribute(
    'href',
    'https://github.com/djordjev/text-cms'
  );

  expect(container).toMatchSnapshot();
});
