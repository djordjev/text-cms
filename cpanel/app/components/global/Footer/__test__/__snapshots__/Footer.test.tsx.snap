// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer 1`] = `
<div>
  <footer
    class="u-footer u-footer-center u-p-2x u-bg-base-300 u-text-base-content"
  >
    <aside
      class="u-block"
    >
      <p
        class="u-inline-block"
      >
        Copyright Â© 2024 - Djordje Vukovic
      </p>
      <div
        class="u-mx-2xs u-inline"
      >
        /
      </div>
      <a
        href="https://github.com/djordjev/text-cms"
        rel="noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </aside>
  </footer>
</div>
`;
